jobs/build/incremental/Jenkinsfile

```diff
@@ -0,0 +1,329 @@
+        }
+    }
+
+    if (ONLY_SCAN.toBoolean() || !images) {
+        currentBuild.description += "\nNo images built."
+        return
+    }
+    filter = "-i ${images.join(',')}"
+
+    /* ************************************************************************** */
+    stage('Rebase images') {
+
+        try {
+            sshagent(['openshift-bot']) {
+
+                buildlib.kinit() // Sets up credentials for dist-git access
+
+                // determine what version to use
+                def version = VERSION_OVERRIDE.trim()
+                def release = RELEASE_OVERRIDE.trim() ? RELEASE_OVERRIDE.trim() : "+"
+
+                if (version == "") {
+                    version = buildlib.doozer(
+                        "${doozer_opts} --quiet images:query-rpm-version --repo-type unsigned",
+                        [capture: true]
+                    ).split(' ').last()
+                } else {
+                    version = version.startsWith("v") ? version : "v${version}"
+                }
+
+                println "Images will be rebased and built at version-release ${version}-${release}"
+                currentBuild.displayName = "#${currentBuild.number} - ${version}-${release}"
+
+                // rebase all images with that version and release
+                buildlib.doozer("""
+                    ${doozer_opts} ${filter} --latest-parent-version
+                    images:rebase
+                    --push --version ${version} --release '${release}'
+                    -m 'Rebuild on upstream change; version ${version}'
+                """)
+            }
+
+        } catch (err) {
+
+            currentBuild.description = "${err} during rebase step"
+            mail_failure(
+                err: err,
+                stage: "Rebase images",
+                extra_body: """Attempting to build updates for:\n${images.join("\n")}""",
+            )
+            // Re-throw the error in order to fail the job
+            throw err
+        } finally {
+            safeArchiveArtifacts(["doozer_working/*.log"])
+        }
+    }
+
+    /* ************************************************************************** */
+    stage('Build images') {
+
+        try {
+
+            buildlib.kinit() // Sets up credentials for brew access
+
+            buildlib.doozer """
+                ${doozer_opts} ${filter}
+                images:build
+                --push-to-defaults --repo-type unsigned
+            """
+
+        } catch (err) {
+
+            def failed_msg = ""
+
+            echo "Error building images: ${err}"
+            currentBuild.description = "${err} during build step"
+
+            failed_builds = buildlib.get_failed_builds(DOOZER_WORKING)
+            if (failed_builds) {
+                failed_msg = "The following build(s) failed:\n"
+                failed_builds.each { img, reason -> failed_msg += "${img}: ${reason} \n" }
+                println failed_msg
```

Comment:
Sometimes you do `echo`, sometimes `println`, why is that?

Reply:
I'm actually a clueless groovy n00b. I don't know if there's even a difference.

Reply:
https://stackoverflow.com/questions/42149652/println-in-call-method-of-vars-foo-groovy-works-but-not-in-method-in-class seems relevant if i could understand it a little better.

I don't think it matters but I'm making them all `echo` for consistency.